pkg/apis/extensions/types.go

```diff
@@ -40,6 +40,14 @@ const (
 	// allowed safe and unsafe sysctls in a pod spec. It's a comma-separated list of plain sysctl
 	// names or sysctl patterns (which end in *). The string "*" matches all sysctls.
 	SysctlsPodSecurityPolicyAnnotationKey string = "security.alpha.kubernetes.io/sysctls"
+
+	// IngressPreSharedCertAnnotationKey represents the specific pre-shared SSL
+	// certicate for the Ingress controller to use. The controller *does not*
+	// manage this certificate, it is the users responsibility to create/delete it.
+	// In GCE, the Ingress controller assigns the SSL certificate with this name
+	// to the target proxies of the Ingress.
+	// TODO: In Nginx.
+	IngressPreSharedCertAnnotationKey = "kubernetes.io/ingress.pre-shared-cert"
```

Comment:
Is this a generic feature (e.g. enough Ingress implementations support it that we will eventually add it to Ingress) or is it GCE/GKE specific?  That changes the name we want here.

If it is GCE specific (which is probably a safe starting assumption), it should be defined in GCE code and probably should be something like `ingress.gcp.kubernetes.io/pre-shared-cert`